File: auth0/src/test/java/com/auth0/android/result/CredentialsMock.java
package com.auth0.android.result;

import androidx.annotation.Nullable;
import java.util.Calendar;
import java.util.Date;
import java.util.TimeZone;

@SuppressWarnings("WeakerAccess")
public class CredentialsMock extends Credentials {

    public static final long CURRENT_TIME_MS = calculateCurrentTime();

    public CredentialsMock(@Nullable String idToken, @Nullable String accessToken, @Nullable String type, @Nullable String refreshToken, @Nullable Long expiresIn) {
        super(idToken, accessToken, type, refreshToken, expiresIn);
    }

    public CredentialsMock(@Nullable String idToken, @Nullable String accessToken, @Nullable String type, @Nullable String refreshToken, @Nullable Date expiresAt, @Nullable String scope) {
        super(idToken, accessToken, type, refreshToken, expiresAt, scope);
    }

    @Override
    long getCurrentTimeInMillis() {
        return CURRENT_TIME_MS;
    }

    private static long calculateCurrentTime() {
        Calendar cal = Calendar.getInstance();
        cal.setTimeZone(TimeZone.getTimeZone("UTC"));
        return cal.getTimeInMillis();
    }
}

File: auth0/src/test/java/com/auth0/android/result/CredentialsMock.kt
package com.auth0.android.result

import java.util.*

public class CredentialsMock {

    public companion object {
        @JvmField
        public val CURRENT_TIME_MS: Long = calculateCurrentTime()

        @JvmField
        public val ONE_HOUR_AHEAD_MS: Long = CURRENT_TIME_MS + 60 * 60 * 1000

        private fun calculateCurrentTime(): Long {
            val cal = Calendar.getInstance()
            cal.timeZone = TimeZone.getTimeZone("UTC")
            return cal.timeInMillis
        }

        public fun create(
            idToken: String,
            accessToken: String,
            type: String,
            refreshToken: String?,
            expiresAt: Date,
            scope: String?
        ): Credentials {
            return Credentials(idToken, accessToken, type, refreshToken, expiresAt, scope)
        }
    }
}